### PR TITLE
Fix twig deprecations

### PR DESCRIPTION
**Pull request by jvasseur**

### CODE REVIEW

File: src/Knp/DictionaryBundle/Twig/DictionaryExtension.php [hunk 1]
```diff
@@ -25,7 +25,7 @@ public function __construct(DictionaryRegistry $dictionaries)
     public function getFunctions()
     {
         return array(
-            'dictionary' => new \Twig_Function_Method($this, 'getDictionary'),
+            new \Twig_SimpleFunction('dictionary', [$this, 'getDictionary']),
```
Comment (PedroTroller) [inline src/Knp/DictionaryBundle/Twig/DictionaryExtension.php]: The current lowest version of php is php5.3, so you have to use long array syntax for the moment :smile:

File: src/Knp/DictionaryBundle/Twig/DictionaryExtension.php [hunk 2]
```diff
@@ -35,7 +35,7 @@ public function getFunctions()
     public function getFilters()
     {
         return array(
-            'dictionary' => new \Twig_Filter_Method($this, 'getValue'),
+            new \Twig_SimpleFilter('dictionary', [$this, 'getValue']),
```
Comment (PedroTroller) [inline src/Knp/DictionaryBundle/Twig/DictionaryExtension.php]: Same here

**Comment by jvasseur** [comment]
@PedroTroller I removed the usages of short array syntax.

**Comment by gdbonino** [comment]
+1

**Pull request merged**